### PR TITLE
DumpSegment tool.

### LINKED ISSUE #2723: DumpSegment tool
It'd be nice to have a dump-segment tool that you could invoke like:

```
io.druid.cli.Main tools dump-segment segment1 [segment2 segment3...]
```

This would point directly at a segment directory and should dump all rows in the segment as, I dunno, JSON maybe? Would probably do something like:

- __time as ISO8601
- dims as null, string, or array of strings
- numeric metrics as numbers
- complex metrics as base64 encoded versions of the raw bytes? I guess? could be worth special-casing this so folks can dump something more useful. Like for HLL it'd be good to see the header and buckets encoded in some friendly way.

### PR DESCRIPTION
**Pull request by gianm**
Fixes #2723.

### CODE REVIEW

File: processing/src/main/java/io/druid/query/NoopQueryWatcher.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query;
+
+import com.google.common.util.concurrent.ListenableFuture;
+
+public class NoopQueryWatcher implements QueryWatcher
```
Comment (nishantmonu51) [inline processing/src/main/java/io/druid/query/NoopQueryWatcher.java]: minor nit: some tests create a NoopQueryWatcher, would be nice to replace them too. Not a blocker for this PR at all.

**Comment by nishantmonu51** [comment]
👍

**Comment by fjy** [comment]
👍

File: docs/content/operations/dump-segment.md [hunk 2]
```diff
@@ -0,0 +1,62 @@
+---
```
Comment (fjy) [inline docs/content/operations/dump-segment.md]: nothing links to this page. How will people find it?

Comment (gianm) [inline docs/content/operations/dump-segment.md]: we could add it to http://druid.io/libraries.html once released

**Pull request merged**